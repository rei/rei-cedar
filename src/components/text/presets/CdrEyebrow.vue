<script setup lang="ts">
import { useCssModule } from 'vue';
import { baseTextProps } from '../../../types/interfaces';

defineOptions({
  name: 'CdrEyebrow',
});


withDefaults(defineProps<baseTextProps>(), {
  tag: 'span',
});

const baseClass = 'cdr-eyebrow';
const style = useCssModule();
</script>

<template>
  <component
    :is="tag"
    :class="style[baseClass]"
  >
    <slot />
  </component>
</template>

<style module src="./styles/CdrEyebrow.module.scss" lang="scss">
</style>  