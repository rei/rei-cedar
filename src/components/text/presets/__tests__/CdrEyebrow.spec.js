import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import CdrEyebrow from '../CdrEyebrow.vue';


describe('CdrEyebrow', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrEyebrow, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
