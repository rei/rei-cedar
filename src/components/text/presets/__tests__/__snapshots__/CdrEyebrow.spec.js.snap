// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrEyebrow > default configuration > renders correctly 1`] = `
<span
  class="cdr-eyebrow"
>
  
  paragraph
  
</span>
`;
